src/super_gradients/training/transforms/processing.py

```diff
@@ -0,0 +1,183 @@
+from typing import Tuple, List, Union
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import numpy as np
+
+from super_gradients.training.transforms.utils import (
+    rescale_image,
+    rescale_bboxes,
+    shift_image,
+    shift_bboxes,
+    rescale_and_pad_to_size,
+    rescale_xyxy_bboxes,
+    get_center_padding_params,
+)
+
+
+@dataclass
+class ProcessingMetadata(ABC):
+    """Metadata including information to postprocess a prediction."""
+
+
+@dataclass
+class ComposeProcessingMetadata(ProcessingMetadata):
+    metadata_lst: List[Union[None, ProcessingMetadata]]
+
+
+@dataclass
+class DetectionPadToSizeMetadata(ProcessingMetadata):
+    shift_w: float
+    shift_h: float
+
+
+@dataclass
+class RescaleMetadata(ProcessingMetadata):
+    original_size: Tuple[int, int]
+    sy: float
+    sx: float
+
+
+@dataclass
+class DetectionPaddedRescaleMetadata(ProcessingMetadata):
```

Comment:
IMO I find DetectionPaddedRescale transform as a very confusing class - is it padding first, or then rescale or vice versa?
Since we already have `RescaleMetadata` and `DetectionPadToSizeMetadata` we can easily describe equivalent postprocessing transform as a composition of these two steps instead of this one. 
I think this would be more clear in terms of order of applied transformations and would not introduce new [unnecessary] entity.

Reply:
I took it out. I added `DetectionSidePadding` (because the padding of `DetectionPadToSizeMetadata` is center and not `DetectionPaddedRescale`) and also an option `keep_aspect_ratio` in `_Rescale` to preserve the ratio or not (`Rescale` used to it, while `DetectionSidePadding` keeps it).